src/main/java/de/tum/in/www1/artemis/service/metis/conversation/auth/ConversationAuthorizationService.java

```diff
@@ -0,0 +1,43 @@
+package de.tum.in.www1.artemis.service.metis.conversation.auth;
+
+import java.util.Objects;
+
+import javax.persistence.Persistence;
+
+import org.springframework.stereotype.Service;
+
+import de.tum.in.www1.artemis.domain.User;
+import de.tum.in.www1.artemis.repository.UserRepository;
+import de.tum.in.www1.artemis.repository.metis.ConversationParticipantRepository;
+import de.tum.in.www1.artemis.service.AuthorizationCheckService;
+
+@Service
+public class ConversationAuthorizationService {
+
+    protected final ConversationParticipantRepository conversationParticipantRepository;
+
+    protected final UserRepository userRepository;
+
+    protected final AuthorizationCheckService authorizationCheckService;
+
+    protected ConversationAuthorizationService(ConversationParticipantRepository conversationParticipantRepository, UserRepository userRepository,
+            AuthorizationCheckService authorizationCheckService) {
+        this.conversationParticipantRepository = conversationParticipantRepository;
+        this.userRepository = userRepository;
+        this.authorizationCheckService = authorizationCheckService;
+    }
+
+    protected User getUserIfNecessary(User user) {
```

Comment:
Needs javadoc.

Reply:
Added javadoc in this commit: 4acaf46ab01ba755be935993dd4e617f1baf1129